src/s1reader/s1_reader.py

```diff
@@ -727,7 +743,8 @@ def _is_zip_annotation_xml(path: str, id_str: str) -> bool:
 
 def load_bursts(path: str, orbit_path: str, swath_num: int, pol: str = 'vv',
                 burst_ids: list[Union[str, S1BurstId]] = None,
-                flag_apply_eap: bool = True):
+                flag_apply_eap: bool = True,
+                mitigate_az_fmrate_mismatch = True):
```

Comment:
I dont think this actually needs to be an argument here, or in `_burst_from_zip` or `_burst_from_safe_dir`:

The only thing this flag is turning on is whether to run `BurstExtendedCoeffs.from_polynomial_lists`. But there's no reason not to run that. This is a long and verbose flag just to calculate an extra set of polynomial coefficients and store them. The actual function is a user-run function from the Burst object, so it shouldn't matter whether they will possibly run it later when deciding if they are loading the bursts

Reply:
The only change you'll need to do is 
at line 649:
1. make this 
```python
        if len(osv_list) > 0:
            # get orbit from state vector list/element tree
            sensing_duration = datetime.timedelta(
                seconds=n_lines * azimuth_time_interval)
            orbit = get_burst_orbit(sensing_start, sensing_start + sensing_duration,
                                    osv_list)
        else:
            orbit = None

```

into this

```python
        sensing_duration = datetime.timedelta(seconds=n_lines * azimuth_time_interval)
        if len(osv_list) > 0:
            # get orbit from state vector list/element tree
            orbit = get_burst_orbit(sensing_start, sensing_start + sensing_duration,
                                    osv_list)
        else:
            orbit = None
```
to avoid the undefined `sensing_duration` error if they don't pass an orbit (which I regularly do to get info about bursts)

Reply:
> I dont think this actually needs to be an argument here, or in `_burst_from_zip` or `_burst_from_safe_dir`:
> 
> The only thing this flag is turning on is whether to run `BurstExtendedCoeffs.from_polynomial_lists`. But there's no reason not to run that. This is a long and verbose flag just to calculate an extra set of polynomial coefficients and store them. The actual function is a user-run function from the Burst object, so it shouldn't matter whether they will possibly run it later when deciding if they are loading the bursts

Agreed. I will make the "extra coefficient loader" to run every time.